modules/smrtrenseq.nf

```diff
@@ -1,42 +1,386 @@
-  seqkit stats $reads > ${reads}.stats
-  """
+    container 'docker://quay.io/biocontainers/seqfu:1.20.3--h1eb128b_2'
+    scratch true
+    cpus 1
+    memory { 1.GB * task.attempt }
+    errorStrategy { task.exitStatus == 137 ? 'retry' : 'finish' }
+    maxRetries 3
+    time '1h'
+    input:
+    path assembly
+    tuple val(sample), path(reads)
+    output:
+    path "${sample}_statistics.txt"
+    publishDir "results/${sample}", mode: 'copy'
+    script:
+    """
+    seqkit stats -b $assembly | sed 's/_assembly\.contigs//g' > ${sample}_statistics.txt
+    """
+}
+
+process ChopSequences {
+    container 'docker://quay.io/biocontainers/meme:5.5.6--pl5321h4242488_0'
+    scratch true
+    cpus 1
+    memory { 2.GB * task.attempt }
+    errorStrategy { task.exitStatus == 137 ? 'retry' : 'finish' }
+    maxRetries 3
+    time '2h'
+    input:
+    path assembly
+    output:
+    path 'chopped.fa'
+    script:
+    """
+    chop_sequences.sh -i $assembly -o chopped.fa
+    """
+}
+
+process NLRParser {
+    container 'docker://quay.io/biocontainers/meme:5.5.6--pl5321h4242488_0'
+    scratch true
+    cpus 2
+    memory { 4.GB * task.attempt }
+    errorStrategy { task.exitStatus == 137 ? 'retry' : 'finish' }
+    maxRetries 3
+    time '8h'
+    input:
+    path chopped
+    output:
+    path 'parser.xml'
+    script:
+    """
+    nlr_parser.sh -t 2 -i $chopped -o parser.xml
+    """
+}
+
+process NLRAnnotator {
+    container 'docker://quay.io/biocontainers/meme:5.5.6--pl5321h4242488_0'
+    scratch true
+    cpus 1
+    memory { 2.GB * task.attempt }
+    errorStrategy { task.exitStatus == 137 ? 'retry' : 'finish' }
+    maxRetries 3
+    time '4h'
+    input:
+    path assembly
+    path parser_xml
+    tuple val(sample), path(reads)
+    val flanking
+    output:
+    path "${sample}_NLR_annotator.txt"
+    path "${sample}_NLR_annotator.fa"
+    publishDir "results/${sample}", mode: 'copy'
+    script:
+    """
+    nlr_annotator.sh -i $parser_xml -o ${sample}_NLR_annotator.txt -f $assembly ${sample}_NLR_annotator.fa $flanking
+    """
+}
+
+process SummariseNLRs {
+    container 'docker://quay.io/biocontainers/python:3.9--1'
+    scratch true
+    cpus 1
+    memory { 1.GB * task.attempt }
+    errorStrategy { task.exitStatus == 137 ? 'retry' : 'finish' }
+    maxRetries 3
+    time '2h'
+    input:
+    path annotator_text
+    tuple val(sample), path(reads)
+    output:
+    path "${sample}_NLR_summary.txt"
+    publishDir "results/${sample}", mode: 'copy'
+    script:
+   """
+    #!/usr/bin/env python3
+    import csv
+
+    contigs = set()
+    count, pseudogenes, genes, complete, complete_pseudogenes = 0, 0, 0, 0, 0
+
+    with open('$annotator_text') as infile:
+        infile_reader = csv.reader(infile, delimiter='\t')
+        for row in infile_reader:
+            count += 1
+            contigs.add(row[0])
+            nlr_type = row[2]
+            if nlr_type == "complete (psuedogene)" or nlr_type == "partial (pseudogene)":
```

Comment:
I've copied this line from your code - is the spelling mistake of pseudogene a part of NLR annotator? I retained it in case

Reply:
Nope that's just a typo that slipped through!

Reply:
This is corrected now